### PR TITLE
Extract filter store from the search store and convert it to Pinia

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #1037 by @obulat

## Description

Search store was responsible for the state of filters, API query parameters, and conversions between them.

This PR extracts the state of filters to a separate store, and converts it to Pinia. The initial filter creation was extracted to `/constants/filters.js` in #1038.
The filter store actions are mostly used by the `searchStore`, and components use its `getters`.

## Testing Instructions

Run the app and see that search and filtering works as usual for all search type, both on the client side and on the server-rendered pages (do a hard page reload, or go directly to an address with a filter checked, such as `http://localhost:8443/search/audio/?q=sound&license=cc0&mature`, try going back using the browser back button.
Note: It won't be possible to test `mature` filter using the UI until #542 is fixed.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by krysal — COMMENTED** [review]
This is looking quite good! The use of getters in components instead of big chunks of code is neat.

Everything seems to work well except for some issues between applied filters:
- When I apply the "Use commercially" filter then the **Science Museum – UK** source filter is disabled.
- Similarly, when I apply the "Modify or adapt" filter several Source filters are disabled, for example, the **Cleveland Museum of Art**, which has some CC0 images so this should not happen.

File: src/stores/filter.js [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+   * @type {import('@nuxtjs/composition-api').ComputedRef<boolean>}
+   */
+  const isAnyFilterApplied = computed(() => {
+    return anyFilterApplied(
+      getMediaTypeFilters({
+        filters,
+        mediaType: searchType.value,
+      })
+    )
+  })
```
[the remark this answers is not in the document]
Comment (obulat) [reply src/stores/filter.js:109]: Thank you! It's much cleaner combined :)

**Comment by obulat** [comment]
> Everything seems to work well except for some issues between applied filters:
> 
> When I apply the "Use commercially" filter then the Science Museum – UK source filter is disabled.
> Similarly, when I apply the "Modify or adapt" filter several Source filters are disabled, for example, the Cleveland Museum of Art, which has some CC0 images so this should not happen.

Thank you for spotting this, @krysal ! I haven't even looked there :) This was caused by the `isFilterDisabled` working for all filterTypes. "Use commercially" was checking for the string "nc" in the filter code, so scie**nc**emuseum got disabled! I changed the function back so that it only checks the licenses and licenseTypes filters.

File: test/unit/specs/components/search-grid-filter.spec.js [hunk 2]
```diff
@@ -124,8 +99,9 @@ describe('SearchGridFilter', () => {
     const uncheckedFilters = screen.queryAllByRole('checkbox', {
       checked: false,
     })
+
     expect(checkedFilters.length).toEqual(0)
     // Filters are reset with the initial `filterData`
-    expect(uncheckedFilters.length).toEqual(25)
+    expect(uncheckedFilters.length).toEqual(24)
```
[the remark this answers is not in the document]
Comment (obulat) [reply test/unit/specs/components/search-grid-filter.spec.js:105]: I don't know, but 24 is correct, I counted all the image filters in the UI (all except for provider filters and mature filters)

**Review by krysal — APPROVED** [review]
Everything seems to be working correctly now. Great work!

**Pull request merged**